components/shell/fish/patches/path_max.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/src/wutil.cpp b/src/wutil.cpp
+index 5b69c47f..9c4bedd2 100644
+--- a/src/wutil.cpp
++++ b/src/wutil.cpp
+@@ -139,7 +139,7 @@ bool wreaddir_for_dirs(DIR *dir, wcstring *out_name) {
+ const wcstring wgetcwd() {
+     wcstring retval;
+ 
+-    char *res = getcwd(NULL, 0);
++    char *res = getcwd(NULL, PATH_MAX);
```

Comment:
@pyhalov from http://pubs.opengroup.org/onlinepubs/009695399/basedefs/limits.h.html:
<pre>
{PATH_MAX}
Maximum number of bytes in a pathname, including the terminating null character.
Minimum Acceptable Value: {_POSIX_PATH_MAX}
[XSI] [Option Start] Minimum Acceptable Value: {_XOPEN_PATH_MAX} [Option End]
</pre>

So I think the terminating null character is included (assuming our strong posix compliance). I haven't found any direct command/proof though.